src/libmongoc/src/mongoc/mongoc-collection-private.h

```diff
@@ -37,6 +37,9 @@ struct _mongoc_collection_t {
    mongoc_read_concern_t *read_concern;
    mongoc_write_concern_t *write_concern;
    bson_t *gle;
+   int64_t timeout_ms;
+   bool timeout_set;
+   mongoc_database_t *parent_db;
```

Comment:
Up until this point, collections have not tracked their parent database objects, if they had one.  Keeping track of that relationship is necessary if we want to have timeoutMS inherit up the hierarchy according to spec.  However, it does introduce a lifetime dependency of collections on their parent databases.

Reply:
Note that collections obtained via `mongoc_client_get_collection` wouldn't even have a parent database object.

Is there any reason not to inherit the timeout value at the time the collection is instantiated (from either the database or client) and not worry about the parent object's state being modified after the fact. AFAIK, that's how RP/RC/WC inheritance works.

Reply:
+1

Reply:
🤔  that would mean that if you later changed the parent database's timeoutMS, the child collection would be stuck with an old value.  There's no rule in the spec about not being able to set the timeoutMUS multiple times, just that you can't reset it to "unset" once it is set.

So, I suppose, do we care if the collection doesn't track changes in the parent db's timeoutMS over time?

Reply:
Changed so that we take the parent db's initial timeout value, but do not track it for changes over time.

Reply:
@samantharitter: I concur. Just like RC/RP/WC, I think it makes sense to inherit at the time of instantiation and not track. Feel free to resolve this thread.